src/error/mod.rs

```diff
@@ -138,6 +186,13 @@ impl<F: ErrorFormatter> Error<F> {
         self.inner.context.get(&kind)
     }
 
+    /// Insert a piece of context
+    #[inline(never)]
+    #[cfg(feature = "error-context")]
+    pub fn insert(&mut self, kind: ContextKind, value: ContextValue) -> Option<ContextValue> {
```

Comment:
The other operations like `with_cmd`, `format`, and `apply` are `fn(self) -> Self` rather than `fn(&mut self)`. Wouldn't it be more consistent to stick to the former style, and allow chaining of building an error with a fixed number of context items?

Reply:
The question is how much should we focus on the builder vs container API.

Unsure but the usability of this section is not a priority as it is a more advanced case to use these.

I'm tempted to leave `insert` as-is but when we add clap plugin support, we'll likely call those builder functions `set` and then we can make an `Error::set` builder function to parallel it.  I wouldn't want to act on this plan until we get to implementing plugin support as I want to see how that will actually turn out and not just my speculation.